[docs] Add documentation for JinaAI service (#118782)
elastic/elasticsearch: docs/reference/inference/service-jinaai.asciidoc

=====
+
.`task_settings` for the `text_embedding` task type
[%collapsible%closed]
=====
`task`:::
(Optional, string)
Specifies the task passed to the model.
Valid values are:
* `classification`: use it for embeddings passed through a text classifier.
* `clustering`: use it for the embeddings run through a clustering algorithm.
* `ingest`: use it for storing document embeddings in a vector database.
* `search`: use it for storing embeddings of search queries run against a vector database to find relevant documents.
=====

[discrete]
[[inference-example-jinaai]]
==== JinaAI service examples

The following example shows how to create {infer} endpoints to get `text_embeddings` and `rerank` and to use them in a search application.

First, we create the `embeddings` service:

[source,console]
------------------------------------------------------------
PUT _inference/text_embedding/jinaai-embeddings
{
    "service": "jinaai",
    "service_settings": {
        "model_id": "jina-embeddings-v3",
        "api_key": "<api_key>",
    },
    "task_settings": {}
}

> ------------------------------------------------------------

Reviewer: ```suggestion
------------------------------------------------------------
// TEST[skip:uses ML]
```

You'll have to skip the snippet testing for console examples because can't test for these. That's what's breaking CI now.

See Watsonx [example](https://github.com/elastic/elasticsearch/blob/8.17/docs/reference/inference/service-watsonx-ai.asciidoc?plain=1#L110) for reference

Author: ah ok

==== {api-path-parms-title}

`<inference_id>`::
(Required, string)
include::inference-shared.asciidoc[tag=inference-id]

`<task_type>`::
(Required, string)
include::inference-shared.asciidoc[tag=task-type]
+
--
Available task types:

* `text_embedding`,
* `rerank`.
--

[discrete]
[[infer-service-jinaai-api-request-body]]
==== {api-request-body-title}

`chunking_settings`::
(Optional, object)
include::inference-shared.asciidoc[tag=chunking-settings]

`max_chunking_size`:::
(Optional, integer)
include::inference-shared.asciidoc[tag=chunking-settings-max-chunking-size]

`overlap`:::
(Optional, integer)
include::inference-shared.asciidoc[tag=chunking-settings-overlap]

`sentence_overlap`:::
(Optional, integer)
include::inference-shared.asciidoc[tag=chunking-settings-sentence-overlap]

`strategy`:::
(Optional, string)
include::inference-shared.asciidoc[tag=chunking-settings-strategy]

`service`::
(Required, string)
The type of service supported for the specified task type. In this case, 
`jinaai`.

`service_settings`::
(Required, object)
include::inference-shared.asciidoc[tag=service-settings]
+
--
These settings are specific to the `jinaai` service.
--

`api_key`:::
(Required, string)

> A valid API key of your JinaAI account.
> You can find in:
> https://jina.ai/embeddings/.

Reviewer: ```suggestion
A valid API key for your JinaAI account.
You can find it at https://jina.ai/embeddings/.
```

+
--
include::inference-shared.asciidoc[tag=api-key-admonition]
--

`rate_limit`:::
(Optional, object)

> By default, the `jinaai` service sets the number of requests allowed per minute to `2000`.
> This value is the same for all task types.
> To modify this, set the `requests_per_minute` setting of this object in your service settings:

Reviewer: ```suggestion
The default rate limit for the `jinaai` service is 2000 requests per minute for all task types. 
You can modify this using the `requests_per_minute` setting in your service settings:
```

Just a bit clearer phrasing

+
--
include::inference-shared.asciidoc[tag=request-per-minute-example]

More information about JinaAI's rate limits can be found in https://jina.ai/contact-sales/#rate-limit.
--
+
.`service_settings` for the `rerank` task type
[%collapsible%closed]
=====
`model_id`::

> (Optional, string)

Reviewer: is this optional in practice?

Author: not actually

The name of the model to use for the {infer} task.

> To review the available `rerank` models, refer to the
> https://jina.ai/reranker.

Reviewer: ```suggestion
To review the available `rerank` compatible models, refer to https://jina.ai/reranker.
```

=====

`task_settings`::
(Optional, object)
include::inference-shared.asciidoc[tag=task-settings]
+
.`task_settings` for the `rerank` task type
[%collapsible%closed]
=====
`return_documents`::
(Optional, boolean)
Specify whether to return doc text within the results.

`top_n`::
(Optional, integer)
The number of most relevant documents to return, defaults to the number of the documents.
If this {infer} endpoint is used in a `text_similarity_reranker` retriever query and `top_n` is set, it must be greater than or equal to `rank_window_size` in the query.
=====
+
.`task_settings` for the `text_embedding` task type
[%collapsible%closed]
=====
`task`:::
(Optional, string)
Specifies the task passed to the model.
Valid values are:
* `classification`: use it for embeddings passed through a text classifier.
* `clustering`: use it for the embeddings run through a clustering algorithm.
* `ingest`: use it for storing document embeddings in a vector database.
* `search`: use it for storing embeddings of search queries run against a vector database to find relevant documents.
=====

[discrete]
[[inference-example-jinaai]]
==== JinaAI service examples

> The following example shows how to create {infer} endpoints to get `text_embeddings` and `rerank` and to use them in a search application.

Reviewer: ```suggestion
The following examples demonstrate how to create {infer} endpoints for `text_embeddings` and `rerank` tasks using the JinaAI service and use them in search requests.
```

First, we create the `embeddings` service:

[source,console]
------------------------------------------------------------
PUT _inference/text_embedding/jinaai-embeddings
{
    "service": "jinaai",
    "service_settings": {
        "model_id": "jina-embeddings-v3",

>         "api_key": "<api_key>",

Reviewer: ```suggestion
        "api_key": "<api_key>"
```

A syntax error

    },

>     "task_settings": {}

Reviewer: We have empty `task_settings` in this example, but extensive options are documented in lines 115-141. Consider showing a more realistic example with commonly used settings.

Author: There are no commonly used settigns for embeddings

}
------------------------------------------------------------
// TEST[skip:uses ML]

Then, we create the `rerank` service:
[source,console]
------------------------------------------------------------
PUT _inference/rerank/jinaai-rerank
{
    "service": "jinaai",
    "service_settings": {

>         "api_key": "<API-KEY>",

Reviewer: We have inconsistent API key formatting between examples: '<api_key>' vs '<API-KEY>'. Should standardize on one for consistency.
